src/Pate/Verification/Widening.hs

```diff
@@ -511,6 +593,33 @@ widenValues sym evalFn bundle postD = do
        g <- execGroundFn evalFn e
        return $ PAD.groundToAbsRange (W4.exprType e) g
 
+widenCells ::
+  [Some (PMc.MemCell sym arch)] ->
+  AbstractDomain sym arch v ->
+  EquivM sym arch (WidenResult sym arch v)
+widenCells cells postD = withSym $ \sym -> do
+  newCells <- liftIO $ PEM.fromList sym [ (c, W4.truePred sym) | c <- cells ]
+  let heapDom = PEE.eqDomainGlobalMemory (PAD.absDomEq $ postD)
+  heapDom' <- liftIO $ PEM.intersect sym heapDom newCells
+  let pred' = (PAD.absDomEq postD){ PEE.eqDomainGlobalMemory = heapDom' }
+  let postD' = postD { PAD.absDomEq = pred' }
+  return (Widen WidenEquality (WidenLocs mempty (Set.fromList cells)) postD')
+
+widenRegs ::
+  [Some (MM.ArchReg arch)] ->
+  AbstractDomain sym arch v ->
+  EquivM sym arch (WidenResult sym arch v)
+widenRegs newRegs postD = withSym $ \sym -> do
+  let
+    regs' = foldl
```

Comment:
You probably want `Data.Foldable.foldl'` here - `foldl` is nearly always a nasty space leak